chapter_6/chapter6_6.cpp

```diff
@@ -0,0 +1,79 @@
+//
+// Brief: monetary contribution tracker based on user input
+// Topics: pointers, structures
+//
+
+#include <iostream>
+#include <string>
+
+struct contributor {
+  std::string name;
+  double amount;
+  std::string title;
+};
+
+int main () {
+  using namespace std;
+  int numContributors;
+
+
+  cout << "Please enter the number of contributors: " << endl;
+
+  if (cin >> numContributors) {
+    // flush the newline (**sigh**)
+    cin.get();
+    // dynamically create memory for an array of structures
+    contributor * contributors = new contributor[numContributors];
+
+    for (int i = 0; i < numContributors; ++i) {
+      double amount;
+      string name;
+
+      cout << "Enter the name of contributor #" << (i+1) << ": " << endl;
+      getline(cin, name);
+      cout << "Enter the amount of contributor #" << (i+1) << ": " << endl;
+      cin >> amount;
+
+      // flush the newline yet again (**ARGH!!!**)
```

Comment:
C#/.Net swoops in to solve the little annoyances like this: https://www.dotnetperls.com/console-readline